tests/FsCheck.Test/Gen.fs

```diff
@@ -119,6 +119,12 @@ module Gen =
         Gen.listOfLength length (Gen.constant v)
         |> sample1
         |> ((=) (List.init length (fun _ -> v)))
+
+    [<Property>]
+    let Shuffle (NonEmptySet (xs:Set<int>)) =
+        Gen.shuffle xs
+        |> sample 10
+        |> List.exists ((<>) (Set.toSeq xs))
```

Comment:
> no that isn't right. It should produce a uniform distribution across all possible permutations of the input, including itself. In other words, [0;1] or [1;0] must be equally likely.

That's my interpretation of Fisher-Yates as well :+1: 

Relying on the statistical attribute of a test being exceedingly unlikely to produce a false positive is, I agree, sometimes the only option. I have tests like that in other code bases as well.

For the sake of someone not aware of this discussion thread, however, I think it might be a good idea to leave a comment in the test, to the effect that this property is non-deterministic, and may fail once in a blue moon.

In this way, if anyone clones the repository, runs all tests, and is unlucky enough to witness a false positive, at least there's 'documentation' in place stating that this can happen, and that one should simply run the tests again.